### PR TITLE
Move typings from DefinitelyTyped

### PR DESCRIPTION
**Pull request by Ty3uK**
According to discussion #49

### CODE REVIEW

File: preact/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export const Redirect: FunctionComponent<RedirectProps>;
+
+export interface SwitchProps {
+  location?: string;
+  children: Array<VNode<RouteProps>>;
+}
+export const Switch: FunctionComponent<SwitchProps>;
+
+export interface RouterProps {
+  hook: () => LocationTuple;
```
Comment (molefrog) [inline preact/index.d.ts]: can we make a special type alias for locationhook function? 
```suggestion
  hook: LocationHook;
```

That way `use-location` type is just `LocationHook` and static location type is `(Path) => LocationHook`

Comment (Ty3uK) [reply preact/index.d.ts]: Done

**Comment by Ty3uK** [comment]
> If that's the case we need to make sure all types are copied to preact/ folder during the prepublishOnly command.

Also done

File: package.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
   "scripts": {
     "test": "jest --verbose --coverage",
     "size": "size-limit",
-    "prepublishOnly": "copyfiles index.js matcher.js use-location.js static-location.js \"extra/**\" preact/"
+    "prepublishOnly": "copyfiles index.js matcher.{js,d.ts} use-location.{js,d.ts} static-location.{js,d.ts} \"extra/**\" preact/"
```
Comment (molefrog) [inline package.json]: Love this trick!

Comment (StrayFromThePath) [inline package.json]: `extra` folder not exists

**Comment by Ty3uK** [comment]
Please, don''t release new version immediately after merge this PR. I will create a new PR to adapt typings for `replace` param that added in #52

File: package.json (hunk 2 above)
Comment (molefrog) [reply package.json]: Right, I've removed it in the 2.0 release. @Ty3uK Can you remove `\"extra/**\"` from the command? It is not used anymore.

Comment (Ty3uK) [reply package.json]: @molefrog should I remove extra from copying?

Comment (molefrog) [reply package.json]: Yes.

**Comment by molefrog** [comment]
> Please, don't release new version immediately after merge this PR. I will create a new PR to adapt typings for replace param that added in #52

Got it.

**Comment by StrayFromThePath** [comment]
How to test typings in this repo?

**Comment by molefrog** [comment]
> How to test typings in this repo?

How do you guys usually do that in DT?

**Comment by Ty3uK** [comment]
https://github.com/microsoft/dtslint

**Comment by StrayFromThePath** [comment]
> > How to test typings in this repo?
> 
> How do you guys usually do that in DT?

Need `tsconfig.json` and `.tsx` file for tests. See here : https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/wouter

**Comment by molefrog** [comment]
Alright, testing types seems to be out of scope of this PR. I suggest we merge this and then start thinking about proper linting. 

@Ty3uK Does everything look good on your end?

**Comment by Ty3uK** [comment]
@molefrog yes

**Pull request merged**